homeassistant/components/lgthinq/coordinator.py

```diff
@@ -0,0 +1,153 @@
+"""DataUpdateCoordinator for the LG ThinQ device."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from thinqconnect import ConnectBaseDevice, DeviceType, ThinQApi, ThinQAPIException
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEVICE_TYPE_API_MAP, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """LG Device's Data Update Coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        device_api: ConnectBaseDevice,
+        *,
+        sub_id: str | None = None,
+    ) -> None:
+        """Initialize data coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}_{device_api.device_id}",
+        )
+
+        # For washTower's washer or dryer
+        self.sub_id = sub_id
+
+        # The device name is usually set to 'alias'.
+        # But, if the sub_id exists, it will be set to 'alias {sub_id}'.
+        # e.g. alias='MyWashTower', sub_id='dryer' then 'MyWashTower dryer'.
+        self.device_name = (
+            f"{device_api.alias} {self.sub_id}" if self.sub_id else device_api.alias
+        )
+
+        # The unique id is usually set to 'device_id'.
+        # But, if the sub_id exists, it will be set to 'device_id_{sub_id}'.
+        # e.g. device_id='TQSXXXX', sub_id='dryer' then 'TQSXXXX_dryer'.
+        self.unique_id = (
+            f"{device_api.device_id}_{self.sub_id}"
+            if self.sub_id
+            else device_api.device_id
+        )
+
+        # Get the api instance.
+        self.device_api = device_api.get_sub_device(self.sub_id) or device_api
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Request to the server to update the status from full response data."""
+        try:
+            data = await self.device_api.thinq_api.async_get_device_status(
+                self.device_api.device_id
+            )
+        except ThinQAPIException as exc:
+            raise UpdateFailed(exc) from exc
+
+        # Full response data into the device api.
+        self.device_api.set_status(data)
+        return data
+
+
+async def async_setup_device_coordinator(
+    hass: HomeAssistant, thinq_api: ThinQApi, device: dict[str, Any]
+) -> list[DeviceDataUpdateCoordinator] | None:
+    """Create DeviceDataUpdateCoordinator and device_api per device."""
+    device_id = device.get("deviceId")
+    if not device_id:
+        _LOGGER.error("Failed to setup device: no device id")
+        return None
+
+    device_info = device.get("deviceInfo")
+    if not device_info:
+        _LOGGER.error("Failed to setup device(%s): no device info", device_id)
+        return None
```

Comment:
Unnecessary.
Removed.

Reply:
Actually there is no error case.
But because type is Any, we check 'device_id' and 'device_info' to escape ruff check issue.
If you have any suggestion, please let me know.

Reply:
`device["deviceId"]`

In case there are more places where we do extra stuff to make mypy happy, feel free to ask help. I'd rather help with this than to have checks in the code for stuff that never happens

Reply:
Thank you for your suggestion.
Modified.